controller/readwrite.go

```diff
@@ -449,6 +472,33 @@ func (rw *ReadWrite) runTask(ctx context.Context, d driver.Driver) error {
 	return err
 }
 
+// deleteTask deletes a task from the drivers map and deletes the task's events.
+// If a task is active and running, it will wait until the task has completed before
+// proceeding with the deletion.
+func (rw *ReadWrite) deleteTask(ctx context.Context, name string) error {
+WAIT:
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			// wait for task to not be running
+			if !rw.drivers.IsActive(name) {
+				break WAIT
```

Comment:
Is the use of label and break label absolutely necessary?
I haven't seen it yet in the code base (I haven't looked thoroughly) but it is generally strongly discouraged. Can the same behavior be achieved in some other way?

Reply:
Wasn't aware that labels were not encouraged, thanks for the insight! I can figure something else out, maybe move this whole wait loop to its own function!